lib/request_handler/json_api_data_parser.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require 'request_handler/schema_parser'
+require 'request_handler/error'
+module RequestHandler
+  class JsonApiDataParser < SchemaParser
+    def initialize(data:, schema:, schema_options: {}, included_schemas: {})
```

Comment:
I think we should clarify the naming a bit - also in regard to how a jsonapi payload looks like.
`data` could be confusing as it's one of the top-level jsonapi keys. Here we actually mean the whole jsonapi document.
Therefore I suggest we go with `document` here.